.github/workflows/ci_pipeline.yml

```diff
@@ -204,6 +204,7 @@ jobs:
           . venv-pySDC/bin/activate
           pip install -e /repositories/pySDC
           pip install qmat
+          pip install pytest-isolate-mpi
```

Comment:
Should be added as a test dependency, directly in the pyproject.toml file, I beleive (see https://github.com/Parallel-in-Time/qmat/blob/4919f197bac230a27f8ccea4188645cb53fefd0a/pyproject.toml#L34) ... also, qmat should already be installed when installing pySDC

Reply:
Hence the idea of optional environment defined in the `pyproject.toml` file. You could create a `gusto` one or a `gusto-test`, etc ... and simply install using :

```bash
pip install -e ./repositories/pySDC[gusto]
```

which would install the base dependencies, + the optional ones (cf [here](https://github.com/Parallel-in-Time/qmat/blob/4919f197bac230a27f8ccea4188645cb53fefd0a/.github/workflows/ci_pipeline.yml#L25)). It makes the pipeline way cleaner, and centralize the dependencies in the `pyproject.toml`, which allows a better control for local or CI tests ...

Reply:
We can think of adding MPI dependencies there, but I don't really see the point, tbh. I prefer this way where the setup of the container and what goes into the container is all together. Outside of the testing pipeline people are not going to use the container and will not need the testing thing.

Also, turns out the test is flaky somehow and adding pytest-isolate-mpi did not at all fix it. This is pretty annoying :D

Reply:
Then, my guess is that you can remove this and the `pip install qmat` :stuck_out_tongue: